### PR TITLE
[ENH] Adding tag for categorical support in `X`

### PR DESCRIPTION
**Pull request by Abhay-Lejith**
#### Reference Issues/PRs

Reference umbrella issue : #6109 

#### What does this implement/fix? Explain your changes.

Added a tag to identify capability of estimator to natively support categorical in X.
#### Does your contribution introduce a new dependency? If yes, which one?
No.

#### Did you add any tests for the change?
No.

##### For all contributions
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

### CODE REVIEW

**Comment by Abhay-Lejith** [comment]
Is the default for boolean tags fixed as `False` or does it have to be specified somewhere ?
And is there anything else to be done while adding a tag?

**Comment by Abhay-Lejith** [comment]
@fkiraly @yarnabrina Do I also have to add a class like this?
https://github.com/sktime/sktime/blob/7637d6f93ff146639f53a56e9ad55177b620fcc4/sktime/registry/_tags.py#L601-L639

File: sktime/registry/_tags.py [hunk 1]
```diff
@@ -1826,6 +1826,12 @@ class y_inner_mtype(_BaseTag):
         ("str", ["temporal", "instance"]),
         "whether the splitter splits by time or by instance (panel/hierarchy index)",
     ),
+    (
+        "capability:categorical-in-X",
```
Comment (yarnabrina) [inline sktime/registry/_tags.py:1830]: Please use `_`, not `-`.

File: sktime/registry/_tags.py [hunk 2]
```diff
@@ -1826,6 +1826,12 @@ class y_inner_mtype(_BaseTag):
...
+        "estimator",
```
Comment (yarnabrina) [inline sktime/registry/_tags.py:1831]: As per the design proposal, I thought we are not planning to add support for any estimator immediately, only forecasters. Please let me know if it was a misunderstanding, having support for all will obviously be preferable.

**Review by yarnabrina — CHANGES_REQUESTED** [review]
To your above question, yes, please add similar class to document tag details.

File: sktime/registry/_tags.py (hunk 2 above)
Comment (Abhay-Lejith) [reply sktime/registry/_tags.py:1831]: Yes, support is only for forecasters for now. 
I will change it.

**Comment by Abhay-Lejith** [comment]
Please assign this pr to me as well and let me know if there are any changes to be made.

**Review by yarnabrina — CHANGES_REQUESTED** [review]
The tag is missing here: https://sktime--6704.org.readthedocs.build/en/6704/api_reference/tags.html.

I think you need to add it to `docs/source/conf.py` as well.

https://github.com/sktime/sktime/blob/66538540566c101408e3742d89548d7971912982/docs/source/conf.py#L349-L361

**Comment by Abhay-Lejith** [comment]
> The tag is missing here: https://sktime--6704.org.readthedocs.build/en/6704/api_reference/tags.html.
> 
> I think you need to add it to `docs/source/conf.py` as well.

It was because it wasn't present in `tags.rst` by the way. Not sure what conf.py does but I added it there as well.

**Review by yarnabrina — APPROVED** [review]

**Pull request merged**